tst/EnergyPlus/unit/DXCoils.unit.cc

```diff
@@ -279,3 +279,525 @@ namespace EnergyPlus {
+
+	int const nPLFfPLR2 = 11;
+	pCurve = &PerfCurve( nPLFfPLR2 );
+	pCurve->CurveType = Quadratic;
+	pCurve->Name = "HP_Heat-PLF-fPLR2";
+	pCurve->Coeff1 = 1;
+	pCurve->Coeff2 = 0;
+	pCurve->Coeff3 = 0;
+	pCurve->Var1Min = 0;
+	pCurve->Var1Max = 1;
+	pCurve->CurveMin = 0.7;
+	pCurve->CurveMax = 1;
+	
+	Coil.MSPLFFPLR( 2 ) = nPLFfPLR2;
+	
+	for ( int CurveNum = 1; CurveNum <= NumCurves; ++CurveNum) {
+		PerfomanceCurveData & rCurve = PerfCurve( CurveNum );
+		if ( rCurve.CurveType == BiQuadratic ) {
+			rCurve.ObjectType = CurveType_BiQuadratic;
+			rCurve.InterpolationType = EvaluateCurveToLimits;
+		} else if ( rCurve.CurveType == Quadratic ) {
+			rCurve.ObjectType = CurveType_Quadratic;
+			rCurve.InterpolationType = EvaluateCurveToLimits;
+		}
+	}
+	
+	// Set up inlet air conditions.
+	Coil.InletAirMassFlowRate = Coil.MSRatedAirMassFlowRate( 1 );
+	MSHPMassFlowRateLow = Coil.MSRatedAirMassFlowRate( 1 );
+	MSHPMassFlowRateHigh = Coil.MSRatedAirMassFlowRate( 2 );
+	OutHumRat = 0.002;
+	OutBaroPress = 101325; // sea level
+	Coil.InletAirTemp = 20;
+	Coil.InletAirHumRat = 0.008;
+	Coil.InletAirEnthalpy = PsyHFnTdbW(Coil.InletAirTemp, Coil.InletAirHumRat);
+	
+	// Test high speed
+	Real64 SpeedRatio = 1.0;
+	Real64 CycRatio = 1.0;
+	int SpeedNum = 2;
+	int const FanOpMode = ContFanCycCoil;
+	
+	// Defroster on
+	OutDryBulbTemp = -5.0; // cold
+	CalcMultiSpeedDXCoilHeating( DXCoilNum, SpeedRatio, CycRatio, SpeedNum, FanOpMode );
+	Real64 COPwoDefrost = Coil.MSRatedCOP( SpeedNum ) / ( CurveValue(nEIRfT2, Coil.InletAirTemp, OutDryBulbTemp) * CurveValue(nEIRfFF2, 1));
+	Real64 COPwDefrost = Coil.TotalHeatingEnergyRate / Coil.ElecHeatingPower;
+	EXPECT_LT( COPwDefrost, COPwoDefrost );
+	
+	// Defroster off
+	OutDryBulbTemp = 5.0; // not cold enough for defroster
+	CalcMultiSpeedDXCoilHeating( DXCoilNum, SpeedRatio, CycRatio, SpeedNum, FanOpMode );
+	COPwoDefrost = Coil.MSRatedCOP( SpeedNum ) / ( CurveValue(nEIRfT2, Coil.InletAirTemp, OutDryBulbTemp) * CurveValue(nEIRfFF2, 1));
+	COPwDefrost = Coil.TotalHeatingEnergyRate / Coil.ElecHeatingPower;
+	EXPECT_DOUBLE_EQ( COPwoDefrost, COPwDefrost );
+	
+	// Test low speed
+	SpeedNum = 1;
+
+	// Defroster on
+	OutDryBulbTemp = -5.0; // cold
+	CalcMultiSpeedDXCoilHeating( DXCoilNum, SpeedRatio, CycRatio, SpeedNum, FanOpMode );
+	COPwoDefrost = Coil.MSRatedCOP( SpeedNum ) / ( CurveValue(nEIRfT1, Coil.InletAirTemp, OutDryBulbTemp) * CurveValue(nEIRfFF1, 1));
+	COPwDefrost = Coil.TotalHeatingEnergyRate / Coil.ElecHeatingPower;
+	EXPECT_LT( COPwDefrost, COPwoDefrost );
+	
+	// Defroster off
+	OutDryBulbTemp = 5.0; // not cold enough for defroster
+	CalcMultiSpeedDXCoilHeating( DXCoilNum, SpeedRatio, CycRatio, SpeedNum, FanOpMode );
+	COPwoDefrost = Coil.MSRatedCOP( SpeedNum ) / ( CurveValue(nEIRfT1, Coil.InletAirTemp, OutDryBulbTemp) * CurveValue(nEIRfFF1, 1));
+	COPwDefrost = Coil.TotalHeatingEnergyRate / Coil.ElecHeatingPower;
+	EXPECT_DOUBLE_EQ( COPwoDefrost, COPwDefrost );
+	
+	// Clean up
```

Comment:
@nmerket If you use ```TEST_F( HVACFixture, foo )``` then you no longer need this clean up here. The Fixture will take care of it for you now. Same in the other unit test.